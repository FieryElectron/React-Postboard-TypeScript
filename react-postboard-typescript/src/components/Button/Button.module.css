.button {
	display: inline-block;
	background: #000;
	color: #fff;
	border: none;
	padding: 8px 12px;
	margin: 8px;
	border-radius: 5px;
	cursor: pointer;
	text-decoration: none;
	font-size: 15px;
	font-family: inherit;
}