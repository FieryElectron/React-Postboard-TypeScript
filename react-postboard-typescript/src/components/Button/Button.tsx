import styles from './Button.module.css';
import React from 'react';

import PropTypes from 'prop-types';

type Props = {
    color:string
    text?:string
    onClick?:React.MouseEventHandler
}

const Button = ({ color, text, onClick } : Props) => {
    return (
        <button data-testid='button' onClick={onClick} style={{ backgroundColor: color }} className={styles.button}>{text}</button>
    )
}

Button.defaultProps = {
    color: '#0071FE',
    text: 'text',
}

Button.propTypes = {
    color: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func,
}


export default Button;