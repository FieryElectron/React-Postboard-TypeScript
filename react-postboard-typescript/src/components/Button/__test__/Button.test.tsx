import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Button from '../Button';

describe('Button', () => {
  it('empty test', () => {

  });
  
  it('renders Button without crashing', () => {
      render(<Button />);
  });
  
  it('renders Button color correctly', () => {
    render(<Button color='red' />);
  
    const button = document.querySelector('[data-testid=button]')!;
    expect(button).toHaveStyle({ 'background-color': 'red' });
  });
  
  it('renders Button color correctly', () => {
    render(<Button text='click me please' />);
  
    const button = document.querySelector('[data-testid=button]')!;
    expect(button).toHaveTextContent('click me please');
  });

  it('should call onClick callback', async () => {
    const onClick = jest.fn();

    render(<Button onClick={onClick} />);

    const button = document.querySelector('[data-testid=button]')!;
    userEvent.click(button);

    await waitFor(() => {
      expect(onClick).toHaveBeenCalled();
    });
  });

  it('matchs snapshot 1', () => {
    const onClick = jest.fn();
    const tree = renderer.create(<Button color='blue' text='click' onClick={onClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('matchs snapshot 2', () => {
    const onClick = jest.fn();
    const tree = renderer.create(<Button color='green' text='save' onClick={onClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});



