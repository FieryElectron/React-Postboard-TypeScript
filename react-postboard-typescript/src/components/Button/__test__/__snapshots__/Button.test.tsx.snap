// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matchs snapshot 1 1`] = `
<button
  className="button"
  data-testid="button"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  click
</button>
`;

exports[`Button matchs snapshot 2 1`] = `
<button
  className="button"
  data-testid="button"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  save
</button>
`;
